client/ayon_core/hosts/max/plugins/publish/collect_members.py

```diff
@@ -12,7 +12,9 @@ class CollectMembers(pyblish.api.InstancePlugin):
     hosts = ['max']
 
     def process(self, instance):
-        if instance.data["productType"] == "workfile":
+        if instance.data["productType"] == (
+            "workfile" or "tycache" or "tyspline"
+        ):
```

Comment:
This comparison is not doing what you think it's doing. This is essentially doing:

```python
if instance.data["productType"] == True
```

Because:
```python
"workfile" or "tycache" or "tyspline" == True
```

```suggestion
        if instance.data["productType"] in {
            "workfile", "tycache", "tyspline"
        }:
```